public/app/core/table_model.ts

```diff
@@ -16,26 +16,16 @@ export default class TableModel {
       return;
     }
 
+    var sign = options.desc ? -1 : 1;
     this.rows.sort(function(a, b) {
       a = a[options.col];
       b = b[options.col];
-      if (a < b) {
-        return -1;
-      }
-      if (a > b) {
-        return 1;
-      }
-      return 0;
+      // Sort null or undefined seperately from comparable values
+      return sign * (+(a == null) - +(b == null) || +(a > b) || -(a < b));
     });
 
     this.columns[options.col].sort = true;
-
-    if (options.desc) {
-      this.rows.reverse();
```

Comment:
wont this disable the reverse sorting?

Reply:
Moved that to `sign` within the sort itself, but perhaps clear to just leave as this reverse, the perf is as good or better.

Reply:
yes, but sign was only used in the equal condition, which looked stange.

Reply:
I've switched it back, but as could be seen from the tests both versions are equivalent.

Reply:
you can change back to using sign, it was just me who misread the diff.

Reply:
The reverse spelling is slightly better I think. Though, all pretty much the same:

https://jsperf.com/sorting-arrays-with-nulls/1